assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js

```diff
@@ -17,6 +19,15 @@ export const CarouselHeaderFrontend = ({ slides, carousel_autoplay }) => {
     autoplayCancelled
   } = useSlides(slidesRef, slides.length - 1, containerRef);
 
+  const changeSlideHandlers = [
```

Comment:
Can you describe why you needed to move this out of the effect?

Reply:
Because I now also use them in the component [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/611/files#diff-98056c746603e53cf86f02263ac85a5a01fc28af6153a5a5299f05811d36ee1eR87)

Reply:
Maybe we should first try to reduce the amount of times these listeners get passed from component to component? :thinking: Currently it's a bit out of control :sweat_smile: 

https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/useSlides.js#L134-L135
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js#L13-L14
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js#L85-L86
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/CarouselHeaderStaticContent.js#L22-L23
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/SlidesContainer.js#L30-L31
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/ArrowsAndIndicators.js#L14-L15
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/Arrows.js#L5-L5

Reply:
With the current changes, the `button` would have a class of `carousel-control-prev`, and a handler called `goToPrevSlide`, while it would actually be going to the next slide.

Reply:
This one would also be incorrect https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/36aae1ed401a88abe5570b54a9c4e29094125846/assets/src/blocks/CarouselHeader/Arrows.js#L7-L7

Reply:
Wouldn't it make more sense to keep the code and markup as is (at least the part for the buttons, the hammer listeners change is probably good), and then ensure through CSS that the buttons are positioned correctly?